core/client/src/main/java/alluxio/client/block/BlockWorkerClient.java

```diff
@@ -378,11 +379,7 @@ public Boolean call() throws AlluxioTException, TException {
         }
       });
     } catch (AlluxioException e) {
-      if (e instanceof WorkerOutOfSpaceException) {
-        return false;
-      } else {
-        throw new IOException(e);
-      }
+      throw new IOException(e);
```

Comment:
probably add a comment here stating this assumption on worker side handling `WorkerOutOfSpaceException`---so in the future if we change worker behavior for same reason, we will adjust code here correspondingly.